monkey/monkey_island/cc/resources/agent_signals.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from http import HTTPStatus
+from json import JSONDecodeError
+
+from flask import request
+
+from monkey_island.cc.event_queue import IIslandEventQueue, IslandEventTopic
+from monkey_island.cc.models import AgentSignals as Signal
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+
+logger = logging.getLogger(__name__)
+
+
+class AgentSignals(AbstractResource):
+    urls = ["/api/agent-signals/terminate-all", "/api/agent-signals/<string:agent_id>"]
+
+    def __init__(
+        self,
+        island_event_queue: IIslandEventQueue,
+    ):
+        self._island_event_queue = island_event_queue
+
+    def post(self):
+        try:
+            signal = Signal(**request.json)
+
+            # We allow an empty timestamp. However, should the agent be able to send us one?
+            if signal.terminate is None:
+                raise ValueError
+            self._island_event_queue.publish(IslandEventTopic.TERMINATE_AGENTS, signal=signal)
```

Comment:
You can just publish the timestamp. `AgentSignals` will have multiple signals in the future, so publishing all signals with a `TERMINATE_AGENTS` event might be confusing.

You probably don't need to parse `request.json` into an `AgentSignal` above. You just need to get the timestamp out of the request and publish it.

Reply:
FYI, right now, the JSON looks like this: `{"kill_time": <timestamp>}`.